compiler-core/src/type_/environment.rs

```diff
@@ -307,14 +307,16 @@ impl<'a> Environment<'a> {
         name: &str,
     ) -> Result<&ValueConstructor, UnknownValueConstructorError> {
         match module {
-            None => {
-                self.local_values
-                    .get(name)
-                    .ok_or_else(|| UnknownValueConstructorError::Variable {
+            None => match self.local_values.get(name) {
```

Comment:
Perhaps. I added this because the following program was failing to compile:
```gleam
fn int_identity(i: Int) -> Int { i }
const f = int_identity
pub fn main() -> Int {
    f(42)
    ^ error here
}
```
The error originated inside `get_value_constructor`, where apparently `f` was not in `self.local_values`. Going up the stack, `get_value_constructor` was called by `get_field_map` which was called by `do_infer_call_with_known_fn`, called by `do_infer_call`. Inside `do_infer_call`, `f` is successfully inferred because `infer_value_constructor` searches the environment using both `get_variable` *and* `get_module_const`. I therefore assumed an additional check needed to be added to `get_value_constructor` to look in `module_values`. Is this a bug?

Reply:
While I would love to help, it might be better for you to do this refactoring.

Reply:
Righto, I'll try and get to this soon.

Reply:
@e-hat all done! https://github.com/gleam-lang/gleam/commit/49ef2769629db3316dcd225aef2fa7cd4d6b324e